pkg/api/persistentvolume/util_test.go

```diff
@@ -28,68 +28,18 @@ import (
 )
 
 func TestDropDisabledFields(t *testing.T) {
-	specWithMode := func(mode *api.PersistentVolumeMode) *api.PersistentVolumeSpec {
-		return &api.PersistentVolumeSpec{VolumeMode: mode}
-	}
-
 	secretRef := &api.SecretReference{
 		Name:      "expansion-secret",
 		Namespace: "default",
 	}
 
-	modeBlock := api.PersistentVolumeBlock
-
 	tests := map[string]struct {
 		oldSpec             *api.PersistentVolumeSpec
 		newSpec             *api.PersistentVolumeSpec
 		expectOldSpec       *api.PersistentVolumeSpec
 		expectNewSpec       *api.PersistentVolumeSpec
-		blockEnabled        bool
 		csiExpansionEnabled bool
 	}{
-		"disabled block clears new": {
-			blockEnabled:  false,
-			newSpec:       specWithMode(&modeBlock),
-			expectNewSpec: specWithMode(nil),
-			oldSpec:       nil,
-			expectOldSpec: nil,
-		},
-		"disabled block clears update when old pv did not use block": {
-			blockEnabled:  false,
-			newSpec:       specWithMode(&modeBlock),
-			expectNewSpec: specWithMode(nil),
-			oldSpec:       specWithMode(nil),
-			expectOldSpec: specWithMode(nil),
-		},
-		"disabled block does not clear new on update when old pv did use block": {
-			blockEnabled:  false,
-			newSpec:       specWithMode(&modeBlock),
-			expectNewSpec: specWithMode(&modeBlock),
-			oldSpec:       specWithMode(&modeBlock),
-			expectOldSpec: specWithMode(&modeBlock),
-		},
-
-		"enabled block preserves new": {
-			blockEnabled:  true,
-			newSpec:       specWithMode(&modeBlock),
-			expectNewSpec: specWithMode(&modeBlock),
-			oldSpec:       nil,
-			expectOldSpec: nil,
-		},
-		"enabled block preserves update when old pv did not use block": {
```

Comment:
I'd probably prefer to leave these tests and toggle their behavior to what we expect now

Reply:
`DropDisabledFields` now does not touch `VolumeMode`, so I find it useless to test it.